cmd/manager/scap.go

```diff
@@ -554,6 +557,11 @@ func filter(ctx context.Context, rawobj []byte, filter string) ([]byte, error) {
 	}
 	if err, ok := v.(error); ok {
 		DBG("Error while filtering: %s", err)
+		// gojq may return a diverse set of internal errors caused by null values.
+		// These errors are happen when a piped filter ends up acting on a null value.
+		if strings.HasSuffix(err.Error(), ": null") {
```

Comment:
I'm not fond the approach of checking the error string sufffix.
But `gojq` may return private error types and we cannot get more insight about the filter value error.
The returned error is not even `gojq.HaltError`, :confused: 
For this specific bug the error returned is `gojq.iteratorError`.

Reply:
Examples of errors I have seen:
```
Fetching URI: '/api/v1/namespaces/-/pods?labelSelector=app%3Dkube-controller-manager'
debug: Applying filter '[[.items[0].spec.containers[0].args[] | select(. | match("--root-ca-file") )] | length | if . ==1 then true else false end]' to path '/api/v1/namespaces/-/pods?labelSelector=app%3Dkube-controller-manager'
debug: Error while filtering: cannot iterate over: null
debug: Error type:, [*gojq.iteratorError]
```
```
debug: Applying filter '[.items[0].spec.containers[0].command | join(" ")]' to path '/api/v1/namespaces/-/pods?labelSelector=app%3Detcd'
debug: Error while filtering: join(" ") cannot be applied to: null
debug: Error type:, [*gojq.func1TypeError]
debug: Persisting warnings to output file
FATAL:Error fetching resources: couldn't filter '{"kind":"PodList","apiVersion":"v1","metadata":{"resourceVersion":"288952"},"items":[]}': join(" ") cannot be applied to: null
```